Can't create Site Collection in SharePoint 2019

<!--
    Your feedback and support is greatly appreciated, thanks for contributing!

    ISSUE TITLE:
    SPSite: Creating a new site collection in SharePoint 2019 throws an Invalid field name error

    ISSUE DESCRIPTION (this template):
    NOTE! Sensitive information should be obfuscated.

    PLEASE KEEP THE HEADERS.
-->
#### Details of the scenario you tried and the problem that is occurring
   When using SharePointDSC to create a Custom minrole server using the Azure SharePoint 2019 image it throws an error for every Site Collection I try to create. New-SPSite works normally outside of DSC. Site Collections with the same settings can be built using New-SPSite in PowerShell or Central Admin

#### Verbose logs showing the problem
VERBOSE: [2018-11-29 14:11:35Z] [VERBOSE] [SPT019]:                            
[[SPSite]TeamSite] Performing the operation "New-SPSite" on target 
"http://Portal2019.Actualdomainname.com".
VERBOSE: [2018-11-29 14:11:58Z] [ERROR] Invalid field name. 
{cbb92da4-fd46-4c7d-af6c-3128c2a5576e} http://portal2019.Actualdomainname.com 
#### Suggested solution to the issue
This works fine in SharePoint 2013 and 2016.  It only happens in 2019
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
           SPWebApplication SharePointSites
            {
                Name = "SharePoint Sites"
                ApplicationPool = $WebAppPoolName
                ApplicationPoolAccount = $WebPoolManagedAccountCreds.UserName
                AllowAnonymous = $false
                DatabaseName = "SP$($ReleaseYear)_Sites_Content"
                WebAppUrl = $PortalUrl
                Port = 80
                PsDscRunAsCredential = $SharePointSetupUserAccountcreds
                DependsOn = "[SPFarm]CreateSPFarm"
            }
        
            SPWebApplication OneDriveSites
            {
                Name = "OneDrive"
                ApplicationPool = $webAppPoolName
                ApplicationPoolAccount = $WebPoolManagedAccountCreds.UserName
                AllowAnonymous = $false
                DatabaseName = "SP$($ReleaseYear)_Sites_OneDrive" 
                HostHeader = "OneDrive$($ReleaseYear).$($AdDomainName)"
                WebAppUrl = $OneDriveUrl
                Port = 80
                PsDscRunAsCredential = $SharePointSetupUserAccountcreds
                DependsOn = "[SPFarm]CreateSPFarm"
            }

            SPCacheAccounts WebAppCacheAccounts
            {
                WebAppUrl = "http://Portal$($ReleaseYear).$($AdDomainName)"
                SuperUserAlias = "$DomainNetbiosName\$SuperUserAlias"
                SuperReaderAlias = "$DomainNetbiosName\$SuperReaderAlias"
                PsDscRunAsCredential = $SharePointSetupUserAccountcreds
                DependsOn = "[SPWebApplication]SharePointSites"
            }

            SPCacheAccounts OneDriveCacheAccounts
            {
                WebAppUrl = "http://oneDrive$($ReleaseYear).$($AdDomainName)"
                SuperUserAlias = "$DomainNetbiosName\$SuperUserAlias"
                SuperReaderAlias = "$DomainNetbiosName\$SuperReaderAlias"
                PsDscRunAsCredential = $SharePointSetupUserAccountcreds
                DependsOn = "[SPWebApplication]OneDriveSites"
            }

            SPSite TeamSite
            {
                Url = $PortalUrl
                OwnerAlias = "$($DomainNetbiosName)\$($SharePointSetupUserName)"
                Name = "Root Demo Site"
                Template = "STS#0"
                PsDscRunAsCredential = $SharePointSetupUserAccountcreds
                DependsOn = "[SPCacheAccounts]WebAppCacheAccounts"
            }

            SPSite MySiteHost
            {                             
                Url = $OneDriveUrl
                OwnerAlias = "$($DomainNetbiosName)\$($SharePointSetupUserName)"
                Name = "OneDrive"
                Template = "SPSMSITEHOST#0"
                PsDscRunAsCredential = $SharePointSetupUserAccountcreds
                DependsOn = "[SPCacheAccounts]OneDriveCacheAccounts"
            }
        }
```

#### The operating system the target node is running
<!--
I am using the regular SharePoint 2019 VM image in Azure.

OsName               : Microsoft Windows Server 2016 Datacenter
OsOperatingSystemSKU : DatacenterServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 14393.2551.amd64fre.rs1_release.181004-1309
OsLanguage           : en-US
OsMuiLanguages       : {en-US}
-->

#### Version of SharePoint that is used (e.g. SharePoint 2016)
SharePoint 2019
#### Version and build of PowerShell the target node is running
<!--
Name                           Value
----                           -----
PSVersion                      5.1.14393.2515
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.2515
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
-->

#### Version of the DSC module that was used ('dev' if using current dev branch)
SharePoint DSC 3.0.0.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- SPSearchServiceApp
+  - Added additional logging at checking db permissions
 - SPWebAppHttpThrottlingMonitor
   - Added new resource to manage web application Http Throttling Monitor settings
 
@@ -42,6 +44,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - Fixed issue where an error was thrown if the specified RootDirectory didn't exist on the
     current server but did exist on the target server. 
   - Fixed issue with using FQDNs instead of NetBIOS server names.
+- SPSite
+  - Implemented workaround to prevent issue with creating site collections immediately after
+    farm creation (Error "Invalid field name. {cbb92da4-fd46-4c7d-af6c-3128c2a5576e}")
 - SPWorkManagementServiceApp
   - Updated links to Docs instead of old TechNet
 

--- a/SharePointDsc/DSCResources/MSFT_SPSearchServiceApp/MSFT_SPSearchServiceApp.psm1
+++ b/SharePointDsc/DSCResources/MSFT_SPSearchServiceApp/MSFT_SPSearchServiceApp.psm1
@@ -156,20 +156,26 @@ function Get-TargetResource
 
             Write-Verbose -Message "Checking Admin Database"
             $adminDB = $serviceApp.SearchAdminDatabase.Name
+
+            Write-Verbose -Message "Checking Admin Database: $adminDB"
             $farmAccountPermissionsNeedCorrecting = (Confirm-UserIsDBOwner -SQLServer $dbServer `
                     -Database $adminDB `
                     -User $farmAccount) -eq $false
+            Write-Verbose -Message "Farm Account Permissions Need Correcting: $farmAccountPermissionsNeedCorrecting"
 
             Write-Verbose -Message "Checking Analytics reporting Database"
             foreach ($database in $serviceApp.AnalyticsReportingDatabases)
             {
                 $analyticsDB = $database.Name
+
+                Write-Verbose -Message "Checking Analytics reporting Database: $analyticsDB"
                 if ($farmAccountPermissionsNeedCorrecting -eq $false)
                 {
                     $farmAccountPermissionsNeedCorrecting = (Confirm-UserIsDBOwner -SQLServer $dbServer `
                             -Database $analyticsDB `
                             -User $farmAccount) -eq $false
                 }
+                Write-Verbose -Message "Farm Account Permissions Need Correcting: $farmAccountPermissionsNeedCorrecting"
             }
 
             Write-Verbose -Message "Checking Crawl Database(s)"
@@ -179,9 +185,12 @@ function Get-TargetResource
                 {
                     $crawlDB = $database.Database.Name
                     $dbServer = $database.Database.NormalizedDataSource
+
+                    Write-Verbose -Message "Checking Crawl Database: $crawlDB"
                     $farmAccountPermissionsNeedCorrecting = (Confirm-UserIsDBOwner -SQLServer $dbServer `
                             -Database $crawlDB `
                             -User $farmAccount) -eq $false
+                    Write-Verbose -Message "Farm Account Permissions Need Correcting: $farmAccountPermissionsNeedCorrecting"
                 }
             }
 
@@ -192,9 +201,12 @@ function Get-TargetResource
                 {
                     $linksDB = $database.Database.Name
                     $dbServer = $database.Database.NormalizedDataSource
+
+                    Write-Verbose -Message "Checking Links Database: $linksDB"
                     $farmAccountPermissionsNeedCorrecting = (Confirm-UserIsDBOwner -SQLServer $dbServer `
                             -Database $linksDB `
                             -User $farmAccount) -eq $false
+                    Write-Verbose -Message "Farm Account Permissions Need Correcting: $farmAccountPermissionsNeedCorrecting"
                 }
             }
 

--- a/SharePointDsc/DSCResources/MSFT_SPSite/MSFT_SPSite.psm1
+++ b/SharePointDsc/DSCResources/MSFT_SPSite/MSFT_SPSite.psm1
@@ -299,7 +299,46 @@ function Set-TargetResource
         {
             Write-Verbose -Message ("Starting New-SPSite with the following parameters: " + `
                     "$(Convert-SPDscHashtableToString $params)")
-            $site = New-SPSite @params
+            #$site = New-SPSite @params
+
+            $paramsList = $params.GetEnumerator() | Where-Object {
+                $_.Key -ne "Verbose"
+            } | ForEach-Object {
+                if ($_.Key -is [Switch])
+                {
+                    "-$($_.Key):$($_.Value)"
+                }
+                else
+                {
+                    "-$($_.Key) '$($_.Value)'"
+                }
+            }
+            $paramsStr = $paramsList -join " "
+            $installedVersion = Get-SPDscInstalledProductVersion
+            if ($installedVersion.ProductMajorPart -eq 15 -or $installedVersion.ProductBuildPart -le 12999)
+            {
+                $newSPSiteCmd = "Add-PSSnapin Microsoft.SharePoint.PowerShell;"
+            }
+            else
+            {
+                $newSPSiteCmd = "Import-Module SharePointServer -Verbose:`$false -WarningAction SilentlyContinue;"
+            }
+
+            $newSPSiteCmd += "New-SPSite " + $paramsStr
+            Write-Verbose "Running command: $newSPSiteCmd"
+
+            $result = Start-Process -Wait `
+                -NoNewWindow `
+                -FilePath "powershell.exe" `
+                -ArgumentList @("-command", "$newSPSiteCmd") `
+                -PassThru
+            if ($result.ExitCode -ne 0)
+            {
+                throw ("[ERROR] An error during site creation. Exit code '$($result.ExitCode)' " + `
+                        "was returned. Please check ULS log for more info")
+            }
+            $site = Get-SPSite -Identity $params.Url -ErrorAction SilentlyContinue
+
             if ($CreateDefaultGroups -eq $true)
             {
                 $doCreateDefaultGroups = $true

--- a/tests/Unit/SharePointDsc/SharePointDsc.SPSite.Tests.ps1
+++ b/tests/Unit/SharePointDsc/SharePointDsc.SPSite.Tests.ps1
@@ -50,7 +50,7 @@ try
     InModuleScope -ModuleName $script:DSCResourceFullName -ScriptBlock {
         Describe -Name $Global:SPDscHelper.DescribeHeader -Fixture {
             BeforeAll {
-                Invoke-Command -Scriptblock $Global:SPDscHelper.InitializeScript -NoNewScope
+                Invoke-Command -ScriptBlock $Global:SPDscHelper.InitializeScript -NoNewScope
 
                 # Initialize tests
                 try
@@ -185,6 +185,12 @@ try
                     }
 
                     Mock -CommandName Get-SPSite -MockWith { return $null }
+
+                    Mock -CommandName Start-Process -MockWith {
+                        return @{
+                            ExitCode = 0
+                        }
+                    }
                 }
 
                 It "Should return OwnerAlias=Null from the get method" {
@@ -197,7 +203,7 @@ try
 
                 It "Should create a new site from the set method" {
                     Set-TargetResource @testParams
-                    Assert-MockCalled New-SPSite
+                    Assert-MockCalled Start-Process
                 }
             }
 

